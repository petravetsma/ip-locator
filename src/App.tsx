import { useEffect, useState } from 'react'
import './App.css'
import { Map } from './Map'
import { GeolocationService } from './service/GeolocationService'
import { GeolocationData } from './model/GeolocationData';
import { ConditionalRender } from './components/ConditionalRender';
import { InputBar } from './logic/InputBar';
import { IpInfoBlock } from './logic/IpInfoBlock';


export function App() {

  const [geolocationData, setGeolocationData] = useState<GeolocationData>(GeolocationData.default);
  // const [geolocationFailData, setGeolocationFailData] = useState<GeolocationFailData>();
  const [isDataLoaded, setDataLoaded] = useState<boolean>(false);

  const geolocationService = new GeolocationService();

  useEffect(() => {
    geolocationService.getData()
      .then(data => {
        console.log(data);
        setDataLoaded(true);
        if (data.success) {
          setGeolocationData(data)
        } else {
          // setGeolocationFailData(data)
        }
      })
  }, []);

  function handleRequestDataByIp(ip: string) {
    setDataLoaded(false);
    geolocationService.getDataByIp(ip)
      .then(data => {
        console.log(data);
        setDataLoaded(true);
        if (data.success) {
          setGeolocationData(data)
        } else {
          // setGeolocationFailData(data)
        }
      })
  }

  return (
    <ConditionalRender condition={isDataLoaded} alternative={<>Loading...</>}>

      <InputBar handleClick={handleRequestDataByIp} />
      <ConditionalRender condition={geolocationData.success} alternative={<>!</>}>
        <IpInfoBlock ip={geolocationData.ip} location={geolocationData.getUserLocation()} isp={geolocationData.isp} />
      </ConditionalRender>

      <Map userLocation={geolocationData.getUserCoordinate()} />
    </ConditionalRender>
  )
}
