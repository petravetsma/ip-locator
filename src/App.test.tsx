import renderer from 'react-test-renderer'
import { App } from './App'

test('App rendered. Show loader on initial app load', () => {
  const component = renderer.create(
    <App />,
  )
  expect(component.toJSON()).toMatchSnapshot()
})