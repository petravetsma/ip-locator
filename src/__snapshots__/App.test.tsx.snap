// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App rendered. Show loader on initial app load 1`] = `
<div
  className="sc-kpDqfm jIHaRp"
>
  <svg
    height="200px"
    preserveAspectRatio="xMidYMid"
    style={
      {
        "background": "rgb(255, 255, 255)",
        "display": "block",
        "margin": "auto",
        "shapeRendering": "auto",
      }
    }
    viewBox="0 0 100 100"
    width="200px"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <circle
      cx={50}
      cy={50}
      fill="none"
      r={35}
      stroke="#fee715"
      strokeDasharray="164.93361431346415 56.97787143782138"
      strokeWidth={10}
    >
      <animateTransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 50;360 50 50"
      />
    </circle>
  </svg>
</div>
`;
